docs/se/config/09_config-profiles.adoc

```diff
@@ -0,0 +1,306 @@
+        return MyConfigSource.create(metaConfig);
+    }
+
+    @Override
+    public Set<String> supported() {
+        return Collections.singleton(TYPE);
+    }
+}
+----
+
+Register it as a java service loader service
+[source]
+.File `META-INF/services/io.helidon.config.spi.ConfigSourceProvider`
+----
+io.helidon.examples.MyConfigSourceProvider
+----
+
+And in `module-info.java` if using JPMS:
+[source,java]
+.File `module-info.java`
+----
+provides io.helidon.config.spi.ConfigSourceProvider with io.helidon.examples.MyConfigSourceProvider
+----
+
+Now you can use the following profile:
+[source,yaml]
+----
+sources:
+  - type: "system-properties"
+  - type: "environment-variables"
+  - type: "my-type"
+    properties:
+        my-property: "some-value"
+----
+
+Note that it is the `io.helidon.config.AbstractConfigSource` class that provides support for
+polling strategies, change watchers, and retry policies. If you create custom config sources that
+should also offer this support be sure they extend `AbstractConfigSource` and implement appropriate
+SPI interfaces (such as `io.helidon.config.spi.WatchableSource`) to support such features.
+
+==== Support for Custom Polling Strategies, Change Watchers, and Retry Policies
+
+Your config profile can include the set-up for polling strategies, change watchers, and retry
+policies if the config source supports them. Declare them in a way similar to
+how you declare the config sources themselves: by `type` and with
+accompanying `properties`.
+
+.Config Profile Support for Built-in Polling Strategies
+|===
+|Strategy Type |Usage |Properties
+
+|`regular`
+| Periodic polling - See link:{javadoc-base-url-api}/PollingStrategies.html#regular-java.time.Duration-[`PollingStrategies.regular`] method
+|`interval` (`Duration`) - indicating how often to poll; e.g., `PT15S` represents 15 seconds
+
+|===
+
+.Config Profile Support for Built-in Change Watchers
+|===
+|Type |Usage |Properties
+
+|`file`
+| Filesystem monitoring - See link:{javadoc-base-url-api}/PollingStrategies.html#watch-java.nio.file.Path-[`PollingStrategies.watch`] method
+| `initial-delay-millis` - delay between the start of the watcher and first check for changes
+
+|===
+
+.Config Profile Support for Built-in Retry Policies
+|===
+|Policy Type |Usage |Properties
+
+|`repeat`
+|Regularly-scheduled - see link:{javadoc-base-url-api}RetryPolicies.html#repeat-int-[`RetryPolicies.repeat`].
+a|`retries` (`int`) - number of retries to perform +
+
+Optional:
+
+* `delay` (`Duration`) - initial delay between retries
+* `delay-factor` (`double`) - `delay` is repeatedly multiplied by this each retry to compute
+the delay for each successive retry
+* `call-timeout` (`Duration`) - timeout for a single invocation to load the source
+* `overall-timeout` (`Duration`) - total timeout for all retry calls and delays
+|===
+
+To specify a custom polling strategy or custom retry policy, implement the interface
```

Comment:
(Q) implement the interface? is that correct?

Reply:
It is correct, but I clarified it to list the exact interfaces to implement.